modules/core/src/smithy4s/http/Metadata.scala

```diff
@@ -211,14 +211,42 @@ object Metadata {
     */
   type Encoder[A] = smithy4s.codecs.Encoder[Metadata, A]
 
-  object Encoder extends CachedEncoderCompilerImpl(awsHeaderEncoding = false) {
+  private final class MetadataEncoder(
+      awsHeaderEncoding: Boolean,
+      explicitDefaultsEncoding: Boolean
+  ) extends CachedEncoderCompilerImpl(
+        awsHeaderEncoding = awsHeaderEncoding,
+        explicitDefaultsEncoding = explicitDefaultsEncoding
+      ) {
     type Compiler = CachedSchemaCompiler[Encoder]
   }
+
+  object Encoder
+      extends CachedEncoderCompilerImpl(
+        awsHeaderEncoding = false,
+        explicitDefaultsEncoding = false
+      ) {
+    type Compiler = CachedSchemaCompiler[Encoder]
+
+    def make(
```

Comment:
not a fan of this. `awsHeaderEncoding` shouldn't be exposed to the users here. 

Also, the UX I'd like would be : `Encoder.withExplicitDefaultsEncoding(true)`. The `CachedEncoderCompilerImpl` class should carry the method, and this `make` method should be removed.